src/txmempool.cpp

```diff
@@ -197,13 +197,29 @@ util::Result<CTxMemPool::setEntries> CTxMemPool::CalculateAncestorsAndCheckLimit
 }
 
 bool CTxMemPool::CheckPackageLimits(const Package& package,
+                                    const int64_t total_vsize,
                                     const Limits& limits,
                                     std::string &errString) const
 {
+    size_t pack_size = package.size();
+
+    // Package itself is busting mempool limits; should be rejected even if no staged_ancestors exist
+    if (pack_size > static_cast<uint64_t>(limits.ancestor_count)) {
```

Comment:
I think those new package limits are debatable in themselves. If a node operator is setting lower `limitancestorcount` / `limitdescendantcount` (e.g 12), a reasonable behavior when we’re accepting over-size package (e.g 25 elements) could be to accept the 12th first component of the package.

I think those new checks are preventing this or at the very least we should document mempool options on the new constraint they’re bringing on package evaluation from my understanding.

Reply:
CheckPackageLimits is only called in "AcceptMultipleTransactions", in other words, we already will try to submit these first parts when they make economic sense.

We should probably do something about the naming of functions at some point to delineate better when they are called, but I'll consider that future work.

Reply:
From my understanding of offline discussions with PR author, those limits will be only considered when the individual transactions are not satisfying the mempool min fee on their own and they serve as an upper bound check on the max number of elements in the transaction.

Yes documentation and naming of functions can be improved in the future.